File: web-frontend/src/app/core/utils/content-disposition.ts
/** Имя файла из заголовка Content-Disposition (filename / filename* UTF-8). */
export function filenameFromContentDisposition(header: string | null | undefined): string | null {
  if (!header || !header.trim()) return null;
  const star = /filename\*\s*=\s*UTF-8''([^;\s]+)/i.exec(header);
  if (star?.[1]) {
    try {
      return decodeURIComponent(star[1].replace(/\+/g, ' '));
    } catch {
      return star[1];
    }
  }
  const plain = /filename\s*=\s*"([^"]+)"/i.exec(header) || /filename\s*=\s*([^;\s]+)/i.exec(header);
  if (plain?.[1]) {
    return plain[1].replace(/^["']|["']$/g, '');
  }
  return null;
}

File: frontend/lib/core/utils/content_disposition.dart
/// Имя файла из заголовка Content-Disposition (filename / filename* UTF-8).
String? filenameFromContentDisposition(String? header) {
  if (header == null || header.trim().isEmpty) return null;
  final star = RegExp(r"filename\*\s*=\s*UTF-8''([^;\s]+)", caseSensitive: false)
      .firstMatch(header);
  if (star != null) {
    try {
      return Uri.decodeComponent(star.group(1)!.replaceAll('+', ' '));
    } catch (_) {
      return star.group(1);
    }
  }
  final plain = RegExp(r'filename\s*=\s*"([^"]+)"', caseSensitive: false)
          .firstMatch(header) ??
      RegExp(r'filename\s*=\s*([^;\s]+)', caseSensitive: false).firstMatch(header);
  if (plain != null) {
    var name = plain.group(1) ?? '';
    if (name.startsWith('"') && name.endsWith('"')) {
      name = name.substring(1, name.length - 1);
    }
    return name.trim().isEmpty ? null : name.trim();
  }
  return null;
}
